### PR TITLE
elbv2 cross zone enabled

### PR DESCRIPTION
**Pull request by abdullahaslam306**

### CODE REVIEW

File: plugins/aws/elbv2/elbv2crosszoneLoadBalancing.js [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+                if (!elbv2Attributes.data.Attributes || !elbv2Attributes.data.Attributes.length){
+                    helpers.addResult(results, 2,
+                        'Application/Network load balancer attributes not found',
+                        region, resource);
+                    return cb();
+                }
+
+                var found = false;
+                elbv2Attributes.data.Attributes.forEach(attribute => {
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2crosszoneLoadBalancing.js:58]: do attributes.find and move helpers.addResult outside loop

File: plugins/aws/elbv2/elbv2crosszoneLoadBalancing.js [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+                                region, resource);
+                        } else {
+                            helpers.addResult(results, 2,
+                                'Load balancer :' + elb.LoadBalancerName + ': does not have cross-zone load balancing enabled',
+                                region, resource);
+                        }
+                    }
+                });
+
+                if (!found) {
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2crosszoneLoadBalancing.js:73]: what is this for?

**Review by AkhtarAmir — CHANGES_REQUESTED** [review]

File: plugins/aws/elbv2/elbv2crosszoneLoadBalancing.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                }
+
+                if (!elbv2Attributes.data.Attributes || !elbv2Attributes.data.Attributes.length){
+                    helpers.addResult(results, 2,
+                        'Application/Network load balancer attributes not found',
+                        region, resource);
+                    continue;
+                }
+
+                let found = elbv2Attributes.data.Attributes.find(attr => attr.Key === 'load_balancing.cross_zone.enabled' && attr.Value === 'true');
```
Comment (alphadev4) [inline plugins/aws/elbv2/elbv2crosszoneLoadBalancing.js:57]:
```suggestion
                let found = elbv2Attributes.data.Attributes.find(attr => attr.Key && attr.Key === 'load_balancing.cross_zone.enabled' && attr.Value && attr.Value === 'true');
```

**Review by alphadev4 — APPROVED** [review]

File: plugins/aws/elbv2/elbv2crosszoneLoadBalancing.spec.js [hunk 4]
```diff
@@ -0,0 +1,223 @@
...
+};
+
+describe('elbv2crosszoneLoadBalancing', function () {
+    describe('run', function () {
+        it('should PASS if load balancer has cross-zone load balancing enabled', function (done) {
+            const cache = createCache([loadBalancers[0]], loadBalancerAttributes[0]);
+            elbv2crosszoneLoadBalancing.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                done();
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2crosszoneLoadBalancing.spec.js:171]:
```suggestion
                expect(results[0].status).to.equal(0);
                expect(results[0].message).to.include('has cross-zone load balancing enabled');
                done();
```

File: plugins/aws/elbv2/elbv2crosszoneLoadBalancing.spec.js [hunk 5]
```diff
@@ -0,0 +1,224 @@
...
+                expect(results[0].message).to.include('has cross-zone load balancing enabled');
+                done();
+            });
+        });
+        
+        it('should FAIL if load balancer does not have cross-zone load balancing enabled', function (done) {
+            const cache = createCache([loadBalancers[0]], loadBalancerAttributes[1]);
+            elbv2crosszoneLoadBalancing.run(cache, {}, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(2);
```
Comment (AkhtarAmir) [inline plugins/aws/elbv2/elbv2crosszoneLoadBalancing.spec.js:180]:
```suggestion
                expect(results[0].status).to.equal(2);
                expect(results[0].message).to.include('does not have cross-zone load balancing enabled');
```

**Review by AkhtarAmir — APPROVED** [review]

**Pull request merged**